### PR TITLE
Prefer injecting tracer in RealTransacter for better testability

### PR DESCRIPTION
**Pull request by wesleyk**
r @rhall @mmihic

### CODE REVIEW

File: misk-testing/src/main/kotlin/misk/testing/MockTracingBackendModule.kt [hunk 1]
```diff
@@ -6,6 +6,7 @@ import misk.inject.KAbstractModule
 
 class MockTracingBackendModule : KAbstractModule() {
   override fun configure() {
-    bind<Tracer>().to<MockTracer>().asEagerSingleton()
+    bind<MockTracer>().asEagerSingleton()
+    bind<Tracer>().to<MockTracer>()
```
Comment (wesleyk) [inline misk-testing/src/main/kotlin/misk/testing/MockTracingBackendModule.kt]: needed to split this out to ensure I could inject a MockTracer and get the same underlying instance

File: misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt [hunk 2]
```diff
@@ -72,8 +73,10 @@ class HibernateModule(
           sessionFactoryProvider.get(), config)
     }).asSingleton()
 
-    bind(transacterKey).toProvider(Provider<Transacter> {
-      RealTransacter(qualifier, sessionFactoryProvider.get(), config)
+    bind(transacterKey).toProvider(object : Provider<Transacter> {
+      @com.google.inject.Inject(optional=true) val tracer: Tracer? = null
```
Comment (ryanhall07) [inline misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt]: do we want to force misk apps to use guice?

Comment (ryanhall07) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt]: so does guice blow up if you try injecting a null value for a nullable type in kt?

Comment (wesleyk) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt]: guice is already all over misk, my impression was that the two are coupled

Comment (wesleyk) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt]: > you try injecting a null value for a nullable type in kt

Yea, tried locally and that's what I get

Comment (ryanhall07) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt]: well is guice impl all over or just javax.Inject? if we decide we want to be opinionated about guice, then we should just use the guice annotations everywhere IMO

Comment (wesleyk) [reply misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt]: it does seem like it's opinionated. There's a class called Guice.

**Pull request merged**